main.go

```diff
@@ -67,13 +71,29 @@ func main() {
 		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "f0fb677d.authorino.3scale.net",
+		LeaderElectionID:       "aac3a15d.authorino.kuadrant.net",
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
+	//get operator namespace
+	namespace := authorinooperatorv1beta1.AuthorinoOperatorNamespace
+	ns, found := os.LookupEnv("WATCH_NAMESPACE")
+	if found {
+		namespace = ns
+	}
+
+	if err = (&controllers.AuthorinoReconciler{
+		Client:    mgr.GetClient(),
+		Log:       ctrl.Log.WithName("authorino-operator").WithName("controller").WithName("Authorino"),
+		Scheme:    mgr.GetScheme(),
+		Namespace: namespace,
```

Comment:
My understanding for "`WATCH_NAMESPACE`" is that this should be the namespace the operator will be watching for `Authorino` CRs (if not the entire cluster). Instead, it here seems to be used as the namespace where the operator is running maybe?

We here seems to be looking for a `ClusterRole` in the namespace that matches the value of this variable, with default to "`authorino-operator`", which doesn't sound right to me. We probably don't even need the name of the namespace to get the `ClusterRole`.

Maybe rename this variable... and either use `WATCH_NAMESPACE` as expected (an option of the manager, not of the controller) or make the operator always cluster-scoped (i.e. get rid of `WATCH_NAMESPACE`).

Reply:
Here's what I had in mind:

```diff
diff --git a/api/v1beta1/authorino_types.go b/api/v1beta1/authorino_types.go
index 93febeb..38abcff 100644
--- a/api/v1beta1/authorino_types.go
+++ b/api/v1beta1/authorino_types.go
@@ -28,9 +28,8 @@ type ConditionType string
 
 const (
 	// ConditionReady specifies that the resource is ready
-	ConditionReady             ConditionType = "Ready"
-	AuthorinoContainerName     string        = "authorino"
-	AuthorinoOperatorNamespace string        = "authorino-operator"
+	ConditionReady         ConditionType = "Ready"
+	AuthorinoContainerName string        = "authorino"
 
 	// Authorino EnvVars
 	ExtAuthGRPCPort         string = "EXT_AUTH_GRPC_PORT"
diff --git a/controllers/authorino_controller.go b/controllers/authorino_controller.go
index 4163185..52d2d68 100644
--- a/controllers/authorino_controller.go
+++ b/controllers/authorino_controller.go
@@ -40,9 +40,8 @@ import (
 // AuthorinoReconciler reconciles a Authorino object
 type AuthorinoReconciler struct {
 	client.Client
-	Namespace string
-	Log       logr.Logger
-	Scheme    *runtime.Scheme
+	Log    logr.Logger
+	Scheme *runtime.Scheme
 }
 
 //+kubebuilder:rbac:groups=operator.authorino.kuadrant.net,resources=authorinoes,verbs=get;list;watch;create;update;patch;delete
@@ -329,8 +328,7 @@ func (r *AuthorinoReconciler) createAuthorinoPermission(authorino *api.Authorino
 
 	authorinoClusterRole := &k8srbac.ClusterRole{
 		ObjectMeta: v1.ObjectMeta{
-			Name:      authorinoClusterRoleName,
-			Namespace: r.Namespace,
+			Name: authorinoClusterRoleName,
 		},
 	}
 	err := r.Get(context.TODO(), client.ObjectKeyFromObject(authorinoClusterRole), authorinoClusterRole)
diff --git a/main.go b/main.go
index a8ce899..589b636 100644
--- a/main.go
+++ b/main.go
@@ -65,31 +65,29 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	managerOptions := ctrl.Options{
 		Scheme:                 scheme,
 		MetricsBindAddress:     metricsAddr,
 		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "aac3a15d.authorino.kuadrant.net",
-	})
+	}
+
+	if ns, found := os.LookupEnv("WATCH_NAMESPACE"); found {
+		managerOptions.Namespace = ns
+	}
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), managerOptions)
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	//get operator namespace
-	namespace := authorinooperatorv1beta1.AuthorinoOperatorNamespace
-	ns, found := os.LookupEnv("WATCH_NAMESPACE")
-	if found {
-		namespace = ns
-	}
-
 	if err = (&controllers.AuthorinoReconciler{
-		Client:    mgr.GetClient(),
-		Log:       ctrl.Log.WithName("authorino-operator").WithName("controller").WithName("Authorino"),
-		Scheme:    mgr.GetScheme(),
-		Namespace: namespace,
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("authorino-operator").WithName("controller").WithName("Authorino"),
+		Scheme: mgr.GetScheme(),
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "Authorino")
 		os.Exit(1)
```

Reply:
> My understanding for "`WATCH_NAMESPACE`" is that this should be the namespace the operator will be watching for `Authorino` CRs (if not the entire cluster). Instead, it here seems to be used as the namespace where the operator is running maybe?
> 
> We here seems to be looking for a `ClusterRole` in the namespace that matches the value of this variable, with default to "`authorino-operator`", which doesn't sound right to me. We probably don't even need the name of the namespace to get the `ClusterRole`.
> 
> Maybe rename this variable... and either use `WATCH_NAMESPACE` as expected (an option of the manager, not of the controller) or make the operator always cluster-scoped (i.e. get rid of `WATCH_NAMESPACE`).

Spot on, we don't need the namespace to fetch the cluster role and as the operator is cluster-scoped we might not need this variable at all